src/Symfony/Component/Translation/Formatter/IntlMessageFormatter.php

```diff
@@ -12,27 +12,50 @@
 namespace Symfony\Component\Translation\Formatter;
 
 use Symfony\Component\Translation\Exception\InvalidArgumentException;
+use Symfony\Component\Translation\Exception\LogicException;
 
 /**
  * @author Guilherme Blanco <guilhermeblanco@hotmail.com>
  * @author Abdellatif Ait boudad <a.aitboudad@gmail.com>
  */
-class IntlMessageFormatter implements MessageFormatterInterface
+class IntlMessageFormatter implements MessageFormatterInterface, IntlFormatterInterface
 {
+    private $hasMessageFormatter;
+    private $cache = array();
+
     /**
      * {@inheritdoc}
      */
     public function format($message, $locale, array $parameters = array())
     {
-        try {
-            $formatter = new \MessageFormatter($locale, $message);
-        } catch (\Throwable $e) {
-            throw new InvalidArgumentException(sprintf('Invalid message format (%s, error #%d).', intl_get_error_message(), intl_get_error_code()), 0, $e);
+        return $this->formatIntl($message, $locale, $parameters);
```

Comment:
should we really format it the Intl way here while we have a new API for that ?

Reply:
OK, removed